tests/test_config_section.py

```diff
@@ -0,0 +1,546 @@
+    'username': 'admin',
+    'password': 'top_secret',
+    'env': '123456',
+    'owner': '123456',
+    'hypervisor_id': 'uuid',
+    'filter_hosts': '*.example.com'
+}
+
+# Values used for testing GlobalConfigSection
+GLOBAL_SECTION_VALUES = {
+    'debug': True,
+    'oneshot': 'true',
+    'print': 'false',
+    'wrong_bool_value': 'nein',
+    'configs': ['local_libvirt.conf'],
+    'log_file': 'my_custom.log',
+    'interval': '120'
+}
+
+
+class TestVirtConfigSection(TestBase):
+    """
+    Test base for testing class VirtConfigSection
+    """
+
+    def setUp(self):
+        """
+        Method executed before each unit test
+        """
+        self.virt_config = VirtConfigSection('test-libvirt', None)
+        # We need to set values using this way, because we need
+        # to trigger __setitem__ of virt_config
+        for key, value in LIBVIRT_SECTION_VALUES.items():
+            self.virt_config[key] = value
+
+    def test_validate_virt_type(self):
+        """
+        Test validation of supported types of virtualization backends
+        """
+        test_virt_types = list(VW_TYPES[:])
+        test_virt_types.extend('vmware,' 'kvm')
+        for virt_type in test_virt_types:
+            self.virt_config['type'] = virt_type
+            result = self.virt_config._validate_virt_type()
+            if virt_type not in VW_TYPES:
+                self.assertIsNotNone(result)
+            else:
+                self.assertIsNone(result)
+                value = self.virt_config.get('type')
+                self.assertEqual(value, virt_type)
+
+    def test_validate_missing_virt_type(self):
+        """
+        Test validation of missing type of virtualization backend
+        """
+        del self.virt_config['type']
+        self.virt_config.validate()
+        virt_type = self.virt_config.get('type')
+        self.assertEqual(virt_type, 'libvirt')
+
+    def test_validate_wrong_virt_type(self):
+        """
+        Test validation of wrong type of virtualization backend
+        """
+        self.virt_config['type'] = 'qemu'
+        result = self.virt_config._validate_virt_type()
+        self.assertIsNotNone(result)
+        self.virt_config.validate()
+        virt_type = self.virt_config.get('type')
+        self.assertEqual(virt_type, 'libvirt')
+
+    def test_validate_unencrypted_password(self):
+        """
+        Test of validation of password that is not encrypted
+        """
+        result = self.virt_config._validate_unencrypted_password('password')
+        self.assertIsNone(result)
+
+    def test_validate_unicode_unencrypted_password(self):
+        """
+        Test of validation of password that is not encrypted and it contains some
+        UTF-8 string.
+        """
+        self.virt_config['password'] = 'Příšerně žluťoučký kůň pěl úděsné ódy.'
+        result = self.virt_config._validate_unencrypted_password('password')
+        self.assertIsNone(result)
+
+    def mock_pwd_file(self):
```

Comment:
I copy pasted this code from: `test_password.py` :-), but you are right. Modification of `Password` globally is not good idea. I added new method `unmock_pwd_file()` to restore `Password` to previous state. What do you think about this workaround?